instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -218,6 +230,32 @@ def _operation_name(self, db_name, statement):
             return self.vendor
         return " ".join(parts)
 
+    def _get_commenter_data(self, conn) -> dict:
+        """Calculate sqlcomment contents from conn and configured options"""
+        commenter_data = {
+            "db_driver": conn.engine.driver,
+            # Driver/framework centric information.
+            "db_framework": f"sqlalchemy:{sqlalchemy.__version__}",
+        }
+
+        if self.commenter_options.get("opentelemetry_values", True):
+            commenter_data.update(**_get_opentelemetry_values())
```

Comment:
Related to this [comment](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2938#issuecomment-2541134100) it seems strange to have trace context related information in a span attribute. I am okay with sqlcommenting as part of the attribute but maybe we can omit the OT values?

Reply:
Capturing Otel context in `db.statement` is a convenience for checking that sqlcommenter is enabled with the current config, and checking that the expected trace context was injected into the query statement sent from db client app to to the db server. The latter is an instrumentation checkup; duplicate `traceparent` can be sqlcommented if both SQLAlchemy (ORM) and db driver (e.g. pymysql/psycopg2) are enabled unintentionally and this could mess with the client-server correlation.

It's not a requirement since all can both be checked in the mysql/postgresql server general logs. But diagnosing issues is quicker with `db.statement` inclusion of full sqlcomment, e.g. looking at span attributes in a local otel-collector exporting to console versus accessing a mysql server and opening its logs. Or e.g. looking at spans exported to a platform versus getting creds and accessing a cloud-hosted db server.